Add owners/approvers aliases for sig-auth subprojects

liggitt opened the pull request:
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

* Adds owner aliases for approvers and reviewers for the sig-auth subprojects.
* References those aliases from owners files in the packages related to that subproject.
* Where there were existing approvers/reviewers, hoisted them into the alias lists.
* Where there were not existing approvers/reviewers (for the audit and encryption-at-rest subprojects), populated lists with kubernetes members in the committers lists relevant to the subproject (added as TMP commits in https://github.com/liggitt/kubernetes/commits/sig-auth-subprojects-commits for inspection)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

xref https://github.com/kubernetes/community/pull/2525

**Special notes for your reviewer**:

OWNERS and COMMITS files generated for each package with https://gist.github.com/liggitt/642a1c5ae1eb16262729a248fe6869ff

```release-note
NONE
```

/hold
for review

will mention on the sig-auth mailing list and in the [subprojects pr](https://github.com/kubernetes/community/pull/2525)

liggitt committed:
audit subproject owners/reviewers

liggitt committed:
authenticators subproject approvers/reviewers

liggitt committed:
authorizers subproject approvers/reviewers

liggitt committed:
certificates subproject approvers/reviewers

liggitt committed:
encryption-at-rest approvers/reviewers

liggitt committed:
node-isolation approvers/reviewers

liggitt committed:
auth policy subproject approvers/reviewers

liggitt committed:
serviceaccount subproject approvers/reviewers

liggitt commented:
/assign @tallclair @enj @mikedanese

liggitt commented:
cc @kubernetes/sig-auth-misc for review
cc @kubernetes/steering-committee for visibility

mikedanese commented on pkg/apis/authentication/OWNERS:
```diff
@@ -1,9 +1,6 @@
-- mbohlool
-- jianhuiz
-- enj
+- sig-auth-authentication-approvers
```
s/authentication/authenticators ?

liggitt replied:
fixed

dims commented:
"added as TMP commits for inspection, will drop before merging" << Big +1

deads2k commented:
I don't see any surprises.

liggitt commented:
moved TMP commits containing commit history to https://github.com/liggitt/kubernetes/commits/sig-auth-subprojects-commits to clean this branch up for merge

added an owners file for `pkg/kubectl/cmd/auth/OWNERS` for the authorizers subproject based on review in https://github.com/kubernetes/community/pull/2525/files#r229072570

as this is intentionally as close to a no-op as possible, would like to get this merged and iterate on the subproject membership

liggitt commented:
/retest

mikedanese commented on OWNERS_ALIASES:
```diff
@@ -1,4 +1,123 @@
+    - wojtek-t
+
+  sig-auth-certificates-approvers:
+    - mikedanese
```
This one is non-alphabetic.

mikedanese commented on OWNERS_ALIASES:
```diff
@@ -1,4 +1,123 @@
+    - deads2k
+    - liggitt
+    - tallclair
+    - mikedanese
```
This one is non-alphabetic.

mikedanese reviewed (commented):
lgtm
/approve

liggitt commented:
> This one is non-alphabetic.

fixed

liggitt commented:
/retest

liggitt commented:
/hold cancel

tallclair commented on plugin/pkg/auth/authorizer/node/OWNERS:
```diff
@@ -1,8 +1,7 @@
-- tallclair
-- liggitt
-- deads2k
+- sig-auth-node-restriction-approvers
```
```suggestion
- sig-auth-node-isolation-approvers
```
Same below.

xref: https://github.com/kubernetes/test-infra/issues/8838#issuecomment-436075246

tallclair reviewed (commented):
Thanks for working on this. Please fix the node-restriction alias, otherwise /lgtm

liggitt commented on OWNERS_ALIASES:
```diff
@@ -1,4 +1,123 @@
+    - lavalamp
+    - liggitt
+    - mbohlool
+    - mml
```
I left these as-is for now, can discuss what we want this to look like (might differ per person)

enj commented:
/lgtm
/approve

I think this is a good place to start.  We can iterate on this as needed.

liggitt commented:
> We suggest the following additional approver: thockin

/assign @thockin
for OWNERS_ALIASES additions
has lgtm from the sig-auth chairs

thockin commented:
/approve
